### PR TITLE
Add folder to the docs-qthelp target (only affects VS)

### LINKED ISSUE #29957: Move docs-qthelp target back into Documentation folder in Visual Studio
As part of the recent work to decouple the docs build from MantidPlot, the docs-qthelp target has been moved out of the Documentation folder in Visual Studio. I discussed this with Martyn and I don't think that was intentional. This issue covers moving it back into that folder. 

@martyngigg - I've created this issue as another small tidy up on the docs. Out of interest why was the original docs-qthelp target split into docs-qtassistant and docs-qthelp? One seems to compile the individual html pages and the other seems to create .qhc\.qch files. Maybe people using Qt Creator can view the output of docs-qtassistant directly without needing to build the docs-qthelp part?

### PR DESCRIPTION
**Pull request by DannyHindson**
**Description of work.**

Move the docs-qthelp target back into the Documentation folder (only affects Visual Studio)

Also fix problem with import readline that was preventing docs build on Windows. I think this module should only be imported on Linux. Have wrapped it in an "if sys.platform" statement that I saw used in qt\applications\workbench\workbench\app\start.py

**To test:**

Run cmake on Windows and check the target shows up in the Documentation folder in Visual Studio:

image

Build the docs-qtassistant target and then the docs-qthelp target and verify that the docs have built OK and look OK when you view the docs in Workbench

Fixes #29957.

*This does not require release notes* because **fixes a non-user problem that only appeared in this release**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by sf1919 — COMMENTED** [review]
The good news is that qt-docs is appearing in VS in Windows. However when I open the help it doesn't display well on my screen. This could be due to my 4k laptop which has caused display problems in the past. I include a screenshot so you can see what I mean.
Screenshot (51)

If the problem was just that qt-docs was missing from VS and help wasn't showing anything at all then those problems have been fixed.

**Comment by DannyHindson** [comment]
> The good news is that qt-docs is appearing in VS in Windows. However when I open the help it doesn't display well on my screen. This could be due to my 4k laptop which has caused display problems in the past. I include a screenshot so you can see what I mean.
> Screenshot (51)
> 
> If the problem was just that qt-docs was missing from VS and help wasn't showing anything at all then those problems have been fixed.

That's interesting! Not seen that before. I'm pretty sure that's not caused by the change in this PR but it's worth looking into as separate issue. Did you build both the docs-qthelp and docs-qtassistant targets when you tested this? Still not sure why that would cause this but thought I'd double check

**Comment by sf1919** [comment]
> 
> That's interesting! Not seen that before. I'm pretty sure that's not caused by the change in this PR but it's worth looking into as >separate issue. Did you build both the docs-qthelp and docs-qtassistant targets when you tested this? Still not sure why that >would cause this but thought I'd double check

Yes I built both. It took a while but I did docs-qtassistant first and then docs-qthelp.

I clicked a few of the links in the help page that opened and they seemed to work so it probably is a separate display issue.

**Comment by DannyHindson** [comment]
> > That's interesting! Not seen that before. I'm pretty sure that's not caused by the change in this PR but it's worth looking into as separate issue. Did you build both the docs-qthelp and docs-qtassistant targets when you tested this? Still not sure why that would cause this but thought I'd double check
> 
> Yes I built both. It took a while but I did docs-qtassistant first and then docs-qthelp.
> 
> I clicked a few of the links in the help page that opened and they seemed to work so it probably is a separate display issue.

OK thanks. Are you happy to approve this PR and we'll pick this up as a separate issue. One of the changes in here fixes a problem with the Windows nightly build not succeeding so would be good to get merged asap.

@martyngigg - have you seen the docs text being small and surrounded by a large border when viewed on a high resolution monitor (see Sarah's screenshot higher up this issue)?

**Review by sf1919 — APPROVED** [review]
docs-qthelp is showing in VS in Windows and help is now working in Workbench.

**Comment by martyngigg** [comment]
@DannyHindson @sf1919 That is odd, I haven't seen that shrinking text issue but I agree it is most likely some kind of 4K resolution. Could we create an issue for it? Feel free to assign it to Core.

File: docs/source/conf.py [hunk 1]
```diff
@@ -11,8 +11,9 @@
 # The workaround is discussed in https://groups.google.com/forum/#!topic/leo-editor/ghiIN7irzY0
 # and simply amounts to importing readline before a QApplication is created in the screenshots
 # directive
-import readline
 import sys
+if sys.platform == "linux" or sys.platform == "linux2" or sys.platform == "darwin":
```
Comment (martyngigg) [inline docs/source/conf.py:15]: Thanks for catching this. I'll stop breaking this for you at some point!

**Comment by DannyHindson** [comment]
> @DannyHindson @sf1919 That is odd, I haven't seen that shrinking text issue but I agree it is most likely some kind of 4K resolution. Could we create an issue for it? Feel free to assign it to Core.

I've created issue https://github.com/mantidproject/mantid/issues/29979 for this. Have assigned it to the current sprint milestone for now but feel free to adjust

**Pull request merged**